### PR TITLE
add multi-batch support

### PR DESCRIPTION
**Pull request by luo-cheng2021**

### CODE REVIEW

File: src/core/shape_inference/include/generate_proposals_shape_inference.hpp [hunk 1]
```diff
@@ -48,32 +48,52 @@ void shape_infer(const GenerateProposalsSingleImage* op,
...
+        }
     }
 
     output_shapes[0] = ov::PartialShape({Dimension::dynamic(), 4});
     output_shapes[1] = ov::PartialShape::dynamic(1);
-    output_shapes[2] = ov::PartialShape({1});
+    if (im_info_shape_rank.is_static()) {
+        output_shapes[2] = ov::PartialShape({im_info_shape[0]});
+    } else {
+        output_shapes[2] = ov::PartialShape::dynamic();
```
Comment (meiyang-intel) [inline src/core/shape_inference/include/generate_proposals_shape_inference.hpp]: ov::PartialShape::dynamic(1) is better?

File: src/core/tests/type_prop/generate_proposals.cpp [hunk 2]
```diff
@@ -111,6 +111,10 @@ TEST(type_prop, generate_proposals_dynamic) {
         ASSERT_EQ(proposals->get_output_element_type(2), element::i64);
         EXPECT_EQ(proposals->get_output_partial_shape(0), (PartialShape{dyn_dim, 4}));
         EXPECT_EQ(proposals->get_output_partial_shape(1), (PartialShape{dyn_dim}));
-        EXPECT_EQ(proposals->get_output_partial_shape(2), (PartialShape{1}));
+        if (s.im_info_shape.rank().is_static()) {
+            EXPECT_EQ(proposals->get_output_partial_shape(2), PartialShape{s.im_info_shape[0]});
+        } else {
+            EXPECT_EQ(proposals->get_output_partial_shape(2), PartialShape::dynamic());
```
Comment (meiyang-intel) [inline src/core/tests/type_prop/generate_proposals.cpp]: PartialShape::dynamic(1)

File: src/plugins/intel_cpu/src/nodes/generate_proposals_single_image.cpp [hunk 3]
```diff
@@ -437,16 +439,26 @@ void GenerateProposalsSingleImageNode::execute(mkldnn::stream strm) {
...
-            uint8_t *p_roi_num_item = reinterpret_cast<uint8_t *>(getChildEdgesAtPort(OUTPUT_ROI_NUM)[0]->getMemoryPtr()->GetPtr());
-
-            auto roi_num_type = getOriginalOutputPrecisionAtPort(OUTPUT_ROI_NUM);
-            fill_output_blobs(&unpacked_boxes[0], &roi_indices_[0], p_roi_item, p_roi_score_item,
-                              p_roi_num_item, pre_nms_topn, num_rois, post_nms_topn_, roi_num_type);
+            int new_num_rois = total_num_rois + num_rois;
+            roi_item.resize(new_num_rois * 4);
+            score_item.resize(new_num_rois);
+
+            p_roi_num += roi_num_type == Precision::I32 ? n * sizeof(int32_t) : n * sizeof(int64_t);
```
Comment (meiyang-intel) [inline src/plugins/intel_cpu/src/nodes/generate_proposals_single_image.cpp]: why the address of 'p_roi_num' is added before fill_output_blobs?

File: src/core/shape_inference/include/generate_proposals_shape_inference.hpp (hunk 1 above)
Comment (luo-cheng2021) [reply src/core/shape_inference/include/generate_proposals_shape_inference.hpp]: ov::PartialShape::dynamic() stands for dynamic rank, dynamic(1) has difference mean

File: src/core/tests/type_prop/generate_proposals.cpp (hunk 2 above)
Comment (luo-cheng2021) [reply src/core/tests/type_prop/generate_proposals.cpp]: ov::PartialShape::dynamic() stands for dynamic rank, dynamic(1) has difference mean

File: src/core/shape_inference/include/generate_proposals_shape_inference.hpp (hunk 1 above)
Comment (luo-cheng2021) [reply src/core/shape_inference/include/generate_proposals_shape_inference.hpp]: fixed.

File: src/core/tests/type_prop/generate_proposals.cpp (hunk 2 above)
Comment (luo-cheng2021) [reply src/core/tests/type_prop/generate_proposals.cpp]: fixed.

File: src/plugins/intel_cpu/src/nodes/generate_proposals_single_image.cpp (hunk 3 above)
Comment (luo-cheng2021) [reply src/plugins/intel_cpu/src/nodes/generate_proposals_single_image.cpp]: fixed.

File: src/frontends/paddle/src/op/generate_proposals_v2.cpp [hunk 4]
```diff
@@ -66,26 +51,31 @@ NamedOutputs generate_proposals_v2(const NodeContext& node) {
...
 
     // generate im_info from im_scale
+    auto batch_index = default_opset::Constant::create<int64_t>(ov::element::i64, {1}, {0});
+    auto batch = std::make_shared<default_opset::Gather>(scores_shape, batch_index, gather_axis);
     auto im_scale = default_opset::Constant::create(ov::element::f32, {1}, {1.0});
-    auto im_info = std::make_shared<default_opset::Concat>(OutputVector{single_im_shape, im_scale}, 0);
+    auto im_scales = std::make_shared<default_opset::Broadcast>(im_scale, batch);
+    auto const_1 = default_opset::Constant::create<int64_t>(ov::element::i64, {1}, {1});
+    auto reshape_im_scales = std::make_shared<default_opset::Unsqueeze>(im_scales, const_1);
+    auto im_info = std::make_shared<default_opset::Concat>(OutputVector{im_shape, reshape_im_scales}, 1);
```
Comment (meiyang-intel) [inline src/frontends/paddle/src/op/generate_proposals_v2.cpp]: It will be more simple if use 'Pad' op to add im_scale into im_info

Comment (luo-cheng2021) [reply src/frontends/paddle/src/op/generate_proposals_v2.cpp]: fixed

**Pull request merged**